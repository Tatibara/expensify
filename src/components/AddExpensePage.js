import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { startAddExpense } from "../actions/expenses";

export class AddExpensePage extends React.Component {
  // export is just for ability to test this unconnected component
  onSubmit = expense => {
    // props.dispatch(addExpense(expense));
    this.props.startAddExpense(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startAddExpense: expense => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
