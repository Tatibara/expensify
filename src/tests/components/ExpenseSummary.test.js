import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should correctly render ExpensesSummary component with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expenseTotal={43430} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render correctly ExpensesSummary component with multiple expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expenseTotal={3000} />
  );
  expect(wrapper).toMatchSnapshot();
});
