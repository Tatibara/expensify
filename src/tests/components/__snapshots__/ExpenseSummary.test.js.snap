// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary component with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling: 
    $434.30
  </h1>
</div>
`;

exports[`should render correctly ExpensesSummary component with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling: 
    $30.00
  </h1>
</div>
`;
